build.analyze.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+# This script runs scan-build to perform a static code analysis
+# https://clang-analyzer.llvm.org/scan-build.html
+#
+# The actual compilers have to be set in the CCC prefix env. variables. e.g:
+# export CCC_CC=clang-3.8
+# export CCC_CXX=clang++3.8
+
+BUILD_DIR=./build.analyze
+REPORT_DIR=report            # relative to BUILD_DIR
+BUILD_WRAPPER="scan-build-3.8"
```

Comment:
Because unfortunately there is no `scan-build`, so otherwise I would have to parse the path and look for the newest version. IMO this is a good trade off as the variable can easily be set.